src/app/ubs/ubs-admin/components/ubs-admin-customers/ubs-admin-customers.component.ts

```diff
@@ -362,13 +364,36 @@ export class UbsAdminCustomersComponent implements OnInit, AfterViewChecked, OnD
     }
   }
 
-  private setColumnWidth(column: any) {
+  private setColumnWidth(column: ColumnParam) {
     const columnEls = Array.from(document.getElementsByClassName('mat-column-' + column.title.key));
     columnEls.forEach((el: any) => {
       el.style.width = column.width + 'px';
     });
   }
 
+  openPopUp(column: ColumnParam, data: string, userId: string): void {
+    this.dialogConfig.disableClose = true;
+    const modalRef = this.dialog.open(CommentPopUpComponent, this.dialogConfig);
+
+    modalRef.componentInstance.header = this.localStorageService.getCurrentLanguage() === 'ua' ? column.title.ua : column.title.en;
+    modalRef.componentInstance.comment = data;
+    modalRef.componentInstance.isLink = true;
+
+    modalRef.afterClosed().subscribe((updatedData: string | null) => {
+      if (updatedData !== null && updatedData !== data) {
+        this.adminCustomerService
+          .addChatLink(userId, updatedData)
+          .pipe(take(1))
+          .subscribe(() => {
+            const rowIndex = this.tableData.findIndex((row) => row.userId === userId);
```

Comment:
The findIndex method is used to locate the row by userId. 
While this works fine, it's not optimal if tableData is large.